### ISSUE TITLE
automodule error with nested BaseModel

### ISSUE THREAD

**Issue by nickeldan** [mansenfranzen/autodoc_pydantic#129]
I have the following MRE:

```python
'''Blah module'''

import typing

import pydantic

class Foo:
    '''Foo class'''

    class Bar(pydantic.BaseModel):
        '''Bar class'''

        x: str = pydantic.Field(description='The x.')

        @pydantic.validator('x')
        def do_nothing(cls, value: typing.Any) -> typing.Any:
            return value
```

In my .rst file, I have

```rst
.. automodule:: blah.blah
    :members:
```

`make html` produces

```
  File ".../autodoc_pydantic/inspection.py", line 352, in __init__
    self.attribute: Dict = self.model.Config
AttributeError: type object 'Foo' has no attribute 'Config'
```

So, `autodoc_pydantic` has somehow been led to believe that `Foo` is a `pydantic.BaseModel`.  What's interesting is, if I remove the validator, then the error goes away.

I'm using Python 3.8.10 and autodoc_pydantic 1.7.1.

**Comment by mansenfranzen** [comment]
Thanks for reporting this bug and the example code. This might be related to #124. I will come back to it soonish.

**Comment by mansenfranzen** [comment]
This bug is addressed via #131.

Before merging the related PR, it would be great if you could test the bugfix on your site. To do so, please install the current dev release in your doc-building-environment via `pip install git+https://github.com/mansenfranzen/autodoc_pydantic.git@v1.7.2-a.1` and rebuild your docs.

**Comment by nickeldan** [comment]
Works perfectly!
